schemas/content/product-attributes.schema.json

```diff
@@ -0,0 +1,101 @@
+{
+  "meta:license": [
+    "Copyright 2018 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/content/product-attributes",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Product Attributes",
+  "type": "object",
+  "meta:extensible": false,
+  "meta:abstract": false,
+  "description": "",
+  "definitions": {
+    "attributes": {
+      "properties": {
+        "xdm:color": {
+          "title": "Color",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:size": {
+          "title": "Size",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:style": {
+          "title": "Style",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:weight": {
+          "title": "Weight",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:fabrication": {
+          "type": "string",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:dimensions": {
+          "title": "Dimensions",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:brand": {
+          "title": "Brand name",
+          "$ref": "#/definitions/attribute"
+        },
+        "xdm:manufacturer": {
+          "title": "Manufacturer name",
+          "$ref": "#/definitions/attribute"
+        }
+      },
+      "patternProperties": {
+        ".+://.+": {
+          "$ref": "#/definitions/attribute"
+        }
+      }
+    },
+    "attribute": {
+      "oneOf": [
+        { "type": "string" },
+        { "type": "number" },
+        { "type": "boolean" },
+        {
+          "type": "object",
+          "title": "Structured Product Attribute",
+          "properties": {
+            "schema:name": {
+              "title": "Name",
+              "type": "string",
+              "description": "The human-readable name of the attribute."
+            },
+            "xdm:value": {
+              "title": "Value",
+              "oneOf": [
+                {
+                  "type": "string"
+                },
+                {
+                  "$ref": "https://ns.adobe.com/xdm/data/measure"
+                }
+              ],
+              "description": "The value of the product attribute."
+            },
+            "xdm:variantAttribute": {
```

Comment:
'xdm:variantAttribute' is not defined for attributes of type "type": "object" only, right?

Reply:
If the attribute is a primitive (string, number, boolean), then `xdm:variantAttribute` should be interpreted as having the default value, i.e. `false`.